### PR TITLE
feat(notification): add callback to Action

### PR DESCRIPTION
**Pull request by pmuir**
* Add an optional callback to the Action. When creating an action the user can optionally specify a callback they would like run
* Add optional callbacks to toast notification list demo
* Export all of notifications

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -36,6 +36,9 @@ export { FilterQuery } from './src/app/filters/filter-query';
 // Notification
 export { ToastNotificationComponent } from './src/app/notification/toast-notification.component';
 export { ToastNotificationListComponent } from './src/app/notification/toast-notification-list.component';
+export { Notification } from './src/app/notification/notification';
+export { NotificationEvent } from './src/app/notification/notification-event';
+export { NotificationModule } from './src/app/notification/notification.module';
```
Comment (joshuawilson) [inline index.ts]: Please add the NotificationModule to the widgets.module.ts too.

**Comment by joshuawilson** [comment]
@pmuir the Tests are failing.

File: index.ts (hunk 1 above)
Comment (pmuir) [inline index.ts]: Done

**Comment by pmuir** [comment]
I think they were failing anyway... The test script was disabled (see additional commit). Pushing that to master.

**Comment by joshuawilson** [comment]
I ran the tests in master and they pass. For this PR, locally I did "npm test" and they failed.

**Pull request merged**